common/src/java/org/apache/hadoop/hive/common/FileUtils.java

```diff
@@ -361,7 +361,7 @@ private static void listS3FilesRecursive(FileStatus base, FileSystem fs, List<Fi
     RemoteIterator<LocatedFileStatus> remoteIterator = fs.listFiles(base.getPath(), true);
     while (remoteIterator.hasNext()) {
       LocatedFileStatus each = remoteIterator.next();
-      Path relativePath = new Path(each.getPath().toString().replace(base.toString(), ""));
+      Path relativePath = new Path(each.getPath().toString().replaceFirst(base.getPath().toString(), ""));
```

Comment:
change makes sense
some ideas:
1. isn't Path.relativize for the same purpose?
a) is so could you please check its source if it does this optimized version or not worse (from performance POV)
b) if it isn't, could you please refactor this to utility method here? in this case, please include unit test into TestFileUtils

Reply:
Thanx @abstractdog for the review.
>isn't [Path.relativize](https://www.geeksforgeeks.org/path-relativize-method-in-java-with-examples/) for the same purpose?



This Path & the path used here is different, the first one is from the Java package. The latter one is from Hadoop-Common
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Path.java



>b) if it isn't, could you please refactor this to utility method here? in this case, please include unit test into TestFileUtils

Done